sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -346,6 +346,41 @@ object RapidsPluginUtils extends Logging {
       loadExtensions(classOf[SparkPlugin], pluginClasses)
     }
   }
+
+  /**
+   * Extracts supported GPU architectures from the given properties file
+   */
+  private def getSupportedGpuArchitectures(propFileName: String): Set[String] = {
+    val props = RapidsPluginUtils.loadProps(propFileName)
+    Option(props.getProperty("gpu_architectures"))
+      .getOrElse(throw new RuntimeException(s"GPU architectures not found in $propFileName"))
+      .split(";")
+      .toSet
+  }
+
+  /**
+   * Checks if the current GPU architecture is supported by the spark-rapids-jni
+   * and cuDF libraries.
+   */
+  def checkGpuArchitecture(): Unit = {
+    val supportedGpuArchsSet = getSupportedGpuArchitectures(JNI_PROPS_FILENAME)
+      .intersect(getSupportedGpuArchitectures(CUDF_PROPS_FILENAME))
+    val majorGpuArch = Cuda.getComputeCapabilityMajor
+    val minorGpuArch = Cuda.getComputeCapabilityMinor
+    // Check if the device's major architecture is supported.
+    val supportedMajorArchs = supportedGpuArchsSet.filter(_.head == majorGpuArch)
```

Comment:
Should we track architecture versions as numbers then, so we can check  

```suggestion
    val supportedMajorArchs = supportedGpuArchsSet.filter(ver => ver / 10 == majorGpuArch)
```

Reply:
Makes sense. This would handles cases with major archs > 10